CHANGES.md

```diff
@@ -1,6 +1,421 @@
+--------------------------
+* fixed a bug in which `make CMAKE=...` did not impact the `cmake` command
+* addressed a fragility in which the compiler build saved a path to a linker
+
+Bug Fixes for GPU Computing
+---------------------------
+* fixed bugs for `locale.name`/`.numPUs` returning bad values on parent locales
+* fixed a bug in which non-normalized CUDA paths were breaking certain flags
+
+Bug Fixes for Libraries
+-----------------------
+
+Bug Fixes for Tools
+-------------------
+* fixed a bug where 'chpldoc' messed up entries following `use`/`import` stmts
+* fixed 'chpldoc' to render string and numerical literals as in source code
+* fixed a bug in which 'chpldoc' was not rendering multi-symbol declarations
+* fixed a 'chpldoc' bug in which nilable classes were mis-rendered
+* fixed some 'chpldoc' bugs in which internal symbols were leaking into docs
+* suppressed a `mason test` hint about using `--show` if was already in use
+
+Platform-specific Bug Fixes
+---------------------------
+* fixed a bug with ABI support on Arm systems when using the LLVM back-end
+
+Third-Party Software Changes
+----------------------------
+* updated the bundled version of Qthreads to version 1.17
+* updated FLTK version from 1.3.5 to 1.3.8 for improved support on Mac M1
+* updated the Python package versions used by 'chpldoc' as follows:
+  - Babel: version 2.10.3
+  - Jinja2: version 3.1.2
+  - MarkupSafe: version 2.1.1
+  - Pygments: version 2.12.0
+  - Sphinx: version 4.5.0
+
+Developer-oriented changes: Process
+-----------------------------------
+
+Developer-oriented changes: Documentation
+-----------------------------------------
+
+Developer-oriented changes: Naming Changes
+------------------------------------------
+
+Developer-oriented changes: Module changes
+------------------------------------------
+* replaced hand-written instability warnings with the `@unstable` annotation
+
+Developer-oriented changes: Performance improvements
+----------------------------------------------------
+
+Developer-oriented changes: Makefile / Build-time changes
+---------------------------------------------------------
+* turned off LLVM assertions for `CHPL_DEVELOPER` builds
+* adjusted generation of `chpl-env-gen.h` to better handle special characters
+
+Developer-oriented changes: Compiler Flags
+------------------------------------------
+* updated `--dyno` to activate the 'dyno' scope resolver
+* added `--warn-int-uint` to warn about implicit `int` to `uint` conversions
+* added `--report-gpu-transform-time` to measure time for GPU transformations
+* added `--library-cmakelists` to generate `cmake` files for a Chapel library
+
+Developer-oriented changes: Compiler improvements/changes
+---------------------------------------------------------
+* removed support for the old, pre-'dyno' parser
+
+Developer-oriented changes: 'dyno' Compiler improvements/changes
+----------------------------------------------------------------
```

Comment:
Here is what I recommend for this section

```
* improved the uAST representation
  - added `uast::ReduceIntent` for some cases that were `uast::Reduce`
  - added escaping for IDs that come from filenames

* made numerous improvements to 'dyno's scope resolution capabilities
  - added support for `this.` and `super.` in `use`/`import` statements
  - added support for `include` statements
  - added support for task intents and reduce intents
  - added support for try-catch statements

* made numerous improvements to 'dyno's type/call resolution capabilities
  - improved support for evaluating `param` expressions..
    (e.g., `1+1` is now resolved to be a `param` whose value is `2`)
  - fixed problems when resolving declarations with both a type and initializer
  - significantly improved return type inference
  - fixed several problems with generic records with defaults
  - fixed a segmentation fault when resolving `extern` routines
  - added support for resolving enums, conditional expressions, and ranges
  - added support for resolving loop index variables and param for-loops
  - added support for resolving varargs argument lists
  - added support for evaluating `==` and `!=` on types

* improved integration with the production compiler
  - can now convert from resolved uAST to `SymExpr`s for the production compiler
  - fixed problems when scope resolving with dyno and using `-M`/`--module-dir`
  - migrated many semantic checks from the production parser to 'dyno'
  - migrated file system capabilities to dyno functions using LLVM libarries

* used LLVM data types to improve performance of parse and scope resolution
* improved the query trace output format and filtering capabilities
```

Reply:
Thanks, this was very helpful!